app/models/coronavirus/sub_section/structured_content.rb

```diff
@@ -0,0 +1,73 @@
+class Coronavirus::SubSection::StructuredContent
+  HEADER_PATTERN = PatternMaker.call(
+    "starts_with hashes then perhaps_spaces then capture(text) and nothing_else",
+    hashes: "#+",
+    text: '\w.+',
+  )
+
+  LINK_PATTERN = PatternMaker.call(
+    "starts_with perhaps_spaces within(sq_brackets,capture(label)) then perhaps_spaces and within(brackets,capture(url))",
+    label: '\s*\w.+',
+    url: '\s*(\b(https?)://)?[-A-Za-z0-9+&@#/%?=~_|!:,.;]+[-A-Za-z0-9+&@#/%=~_|]\s*',
+  )
+
+  Header = Struct.new(:text)
+  Link = Struct.new(:label, :url)
+
+  attr_reader :items
+
+  def self.error_lines(raw_content)
+    return [] unless raw_content
+
+    raw_content.lines.filter_map do |line|
+      stripped = line.strip
+      next if stripped.empty?
+
+      stripped if !HEADER_PATTERN.match(stripped) && !LINK_PATTERN.match(stripped)
+    end
+  end
+
+  def self.parseable?(raw_content)
+    !raw_content.nil? && error_lines(raw_content).empty?
```

Comment:
We've swapped this over to use `present?`. For some reason getting rid of it makes a load of tests fail, I think because some tests used to work without `raw_content`. I think this would be a good thing to fix but I think it might be best to add it in a separate pr.

Reply:
Jon isn't this a method that you've added? Doesn't that mean that you've created the tests? Something somewhere isn't passing the data correctly, so either the tests haven't been set-up correctly, or this points to a larger problem somewhere in the code that really needs to be fixed.

Reply:
As it turns out it's not to do with the test data but rather that a new subsection is created completely blank when the page is loaded. This means that the issue can be seen by removing the `raw_content.present? &&` portion of this line, booting up and trying to create a new sub section. You'll hit a standard rails error because it tries to create a subsection but falls over when it tries to call `lines` on `nil` [here](https://github.com/alphagov/collections-publisher/pull/1324/files#diff-277a4e00cb4c1ba05ddb7f31004e294012a3c761fc5c96527070045d35c1c324R35). 

I don't really understand why this happens but I'll look into it. I don't think it's a problem that we've introduced here but rather something new that this code has exposed.